### PR TITLE
[azure_frontdoor] Fix Azure frontdoor preserve original event

### PR DESCRIPTION
**Pull request by nicpenning**
- Bug

## What does this PR do?
Fixes the "Preserve Original Event" feature for both Frontdoor Access and WAF data streams which has been stated here:
https://github.com/elastic/integrations/issues/6847
https://github.com/elastic/integrations/issues/6848

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

### CODE REVIEW

File: packages/azure_frontdoor/data_stream/access/elasticsearch/ingest_pipeline/default.yml [hunk 1]
```diff
@@ -135,27 +135,22 @@ processors:
...
-
   - remove:
       field:
         - azure.frontdoor.access.properties.securityProtocol
         - azure.frontdoor.access.properties.httpStatusDetails
         - azure.frontdoor.access.time
         - azure.frontdoor.access.properties
-        - event.original
       ignore_missing: true
-
```
Comment (efd6) [inline packages/azure_frontdoor/data_stream/access/elasticsearch/ingest_pipeline/default.yml:158]: Why are these whitespace changes here?

File: packages/azure_frontdoor/data_stream/waf/elasticsearch/ingest_pipeline/default.yml [hunk 2]
```diff
@@ -118,11 +114,16 @@ processors:
...
+      ignore_missing: true
 
 on_failure:
   - set:
       field: event.kind
       value: pipeline_error
   - append:
       field: error.message
-      value: '{{{ _ingest.on_failure_message }}}'
+      value: '{{{ _ingest.on_failure_message }}}'
```
Comment (efd6) [inline packages/azure_frontdoor/data_stream/waf/elasticsearch/ingest_pipeline/default.yml:129]:
```suggestion
      value: '{{{ _ingest.on_failure_message }}}'
```

File: packages/azure_frontdoor/data_stream/waf/elasticsearch/ingest_pipeline/default.yml [hunk 3]
```diff
@@ -83,21 +83,17 @@ processors:
...
         - ISO8601
-
   - remove:
       field:
         - azure.frontdoor.waf.time
         - azure.frontdoor.waf.properties
         - azure.frontdoor.waf.details.matches
-        - event.original
       ignore_missing: true
-
```
Comment (efd6) [inline packages/azure_frontdoor/data_stream/waf/elasticsearch/ingest_pipeline/default.yml:100]: Do we need these whitespace changes?

File: packages/azure_frontdoor/data_stream/access/elasticsearch/ingest_pipeline/default.yml (hunk 1 above)
Comment (nicpenning) [reply packages/azure_frontdoor/data_stream/access/elasticsearch/ingest_pipeline/default.yml:158]: Not sure what the reason for why there were white spaces after some processors but not all so I removed them all.

File: packages/azure_frontdoor/data_stream/waf/elasticsearch/ingest_pipeline/default.yml (hunk 3 above)
Comment (nicpenning) [reply packages/azure_frontdoor/data_stream/waf/elasticsearch/ingest_pipeline/default.yml:100]: We do not need them. Do you know if the format command properly spaces these?

Comment (nicpenning) [reply packages/azure_frontdoor/data_stream/waf/elasticsearch/ingest_pipeline/default.yml:100]: It does not appear to make a difference. Are there guidelines around the required spacing between the processors?

File: packages/azure_frontdoor/data_stream/access/elasticsearch/ingest_pipeline/default.yml [hunk 4]
```diff
@@ -179,11 +174,16 @@ processors:
...
+      ignore_missing: true
 
 on_failure:
   - set:
       field: event.kind
       value: pipeline_error
   - append:
       field: error.message
-      value: '{{{ _ingest.on_failure_message }}}'
+      value: '{{{ _ingest.on_failure_message }}}'
```
[the remark this answers is not in the document]
Comment (nicpenning) [reply packages/azure_frontdoor/data_stream/access/elasticsearch/ingest_pipeline/default.yml:189]: Doesn't look like a change is able to be applied above.

**Comment by nicpenning** [comment]
@efd6 👋

File: packages/azure_frontdoor/data_stream/waf/elasticsearch/ingest_pipeline/default.yml (hunk 3 above)
Comment (efd6) [reply packages/azure_frontdoor/data_stream/waf/elasticsearch/ingest_pipeline/default.yml:100]: No, not really.

File: packages/azure_frontdoor/data_stream/waf/elasticsearch/ingest_pipeline/default.yml [hunk 5]
```diff
@@ -118,6 +114,11 @@ processors:
       field: source.as.organization_name
       target_field: source.as.organization.name
       ignore_missing: true
+  - remove:
+      field: event.original
+      if: "ctx?.tags == null || !(ctx.tags.contains('preserve_original_event'))"
```
Comment (efd6) [inline packages/azure_frontdoor/data_stream/waf/elasticsearch/ingest_pipeline/default.yml:119]:
```suggestion
      if: "ctx.tags == null || !(ctx.tags.contains('preserve_original_event'))"
```

File: packages/azure_frontdoor/data_stream/access/elasticsearch/ingest_pipeline/default.yml [hunk 6]
```diff
@@ -179,6 +174,11 @@ processors:
   - user_agent:
       field: user_agent.original
       ignore_missing: true
+  - remove:
+      field: event.original
+      if: "ctx?.tags == null || !(ctx.tags.contains('preserve_original_event'))"
```
Comment (efd6) [inline packages/azure_frontdoor/data_stream/access/elasticsearch/ingest_pipeline/default.yml:179]:
```suggestion
      if: "ctx.tags == null || !(ctx.tags.contains('preserve_original_event'))"
```

Comment (nicpenning) [reply packages/azure_frontdoor/data_stream/access/elasticsearch/ingest_pipeline/default.yml:179]: Good catch

File: packages/azure_frontdoor/data_stream/waf/elasticsearch/ingest_pipeline/default.yml (hunk 3 above)
Comment (nicpenning) [reply packages/azure_frontdoor/data_stream/waf/elasticsearch/ingest_pipeline/default.yml:100]: Okay, thanks!

**Comment by nicpenning** [comment]
Checking in here. Anything needed here to review?

**Comment by nicpenning** [comment]
Updated, as this got stale. Can someone please review?

File: packages/azure_frontdoor/data_stream/access/elasticsearch/ingest_pipeline/default.yml (hunk 6 above)
Comment (efd6) [reply packages/azure_frontdoor/data_stream/access/elasticsearch/ingest_pipeline/default.yml:180]: This needs to be addressed.

**Comment by nicpenning** [comment]
I think I resolved the issues you addressed.

**Review by efd6 — APPROVED** [review]
Thanks

**Pull request merged**